aws-api/src/main/java/com/amplifyframework/api/aws/AWSGraphQLOperation.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amplifyframework.api.aws
+
+import com.amplifyframework.AmplifyException
+import com.amplifyframework.annotations.InternalAmplifyApi
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.graphql.GraphQLOperation
+import com.amplifyframework.api.graphql.GraphQLRequest
+import com.amplifyframework.api.graphql.GraphQLResponse
+
+/**
+ * A Base AWS GraphQLOperation that also takes an apiName to allow LazyModel support.
+ * @param <R> The type of data contained in the GraphQLResponse.
+ * @param graphQLRequest  A GraphQL request
+ * @param responseFactory an implementation of ResponseFactory
+ * @param apiName to use
+ */
+@InternalAmplifyApi
+abstract class AWSGraphQLOperation<R>(
+    graphQLRequest: GraphQLRequest<R>,
+    responseFactory: GraphQLResponse.Factory,
+    private val apiName: String?
```

Comment:
Yes each api must have a name, but it doesn't mean that apiName was passed to the operation. Its only needed for the new lazy type use cases. The real Operation classes are public and do not require an apiName.

Reply:
Okay so just to confirm, apiName is only required to be supplied to the lazy load operation if it was supplied in the original request. If it was initially omitted it can be omitted from the lazy loads.

Reply:
Correct. However, for all normal purposes, the lazy types should pretty much always have an apiName.

So if a customer uses `AWSApiPlugin` apiName will always be included. For example.
```
@Nullable
@Override
public <R> GraphQLOperation<R> query(
        @NonNull GraphQLRequest<R> graphQLRequest,
        @NonNull Consumer<GraphQLResponse<R>> onResponse,
        @NonNull Consumer<ApiException> onFailure) {
    final String apiName;
    try {
        apiName = getSelectedApiName(EndpointType.GRAPHQL);
    } catch (ApiException exception) {
        onFailure.accept(exception);
        return null;
    }
    return query(apiName, graphQLRequest, onResponse, onFailure);
}
```
See that apiName is grabbed anyway. Its just needs to be optional directly on the operation due to public contract.